### PR TITLE
[IMP] web: allow dynamic placeholder for dates

### PR DESCRIPTION
**Pull request by RicGR98**
See commit

Enterprise PR: https://github.com/odoo/enterprise/pull/72121

### CODE REVIEW

**Comment by aab-odoo** [comment]
Hello @RicGR98 

Could you provide a task id with motivations? I have to say I'm not in favor of adding this new props to all fields now... Moreover, the API is quite restrictive as you enforce the placeholder field to be a date(time) field. Also, if this PR should be accepted, it would require a test and a PR on the documentation repo.

**Comment by RicGR98** [comment]
Hello @aab-odoo :)

> Could you provide a task id with motivations? 

There is no task, just a linked Enterprise PR that motivates this PR: https://github.com/odoo/enterprise/pull/72121

> I have to say I'm not in favor of adding this new props to all fields now... 

Would you rather keep that in accounting perhaps ?

> Moreover, the API is quite restrictive as you enforce the placeholder field to be a date(time) field. 

Yes this was point because we're in a DateTimeField, I'm not sure I see the problem here

> Also, if this PR should be accepted, it would require a test and a PR on the documentation repo.

I can add that indeed, but only for the accounting module then ?

File: addons/web/static/src/views/fields/datetime/datetime_field.js [hunk 1]
```diff
@@ -78,6 +79,15 @@ export class DateTimeField extends Component {
         return ensureArray(this.state.value);
     }
 
+    get placeholder() {
+        const dynamicPlaceholder =
+            this.props.placeholderField && this.props.record.data[this.props.placeholderField];
+        if (dynamicPlaceholder) {
+            return this.getFormattedValue(dynamicPlaceholder);
```
Comment (svfu-odoo) [inline addons/web/static/src/views/fields/datetime/datetime_field.js:86]: Sorry to interject and intrude into the web scope. But I did a quick functional test to check the enterprise PR.
```suggestion
            return this.formatValue(dynamicPlaceholder);
```

Comment (RicGR98) [reply addons/web/static/src/views/fields/datetime/datetime_field.js]: Indeed, thanks!

**Comment by aab-odoo** [comment]
Hello @RicGR98 

We have a task in our pipe to add the support for dynamic placeholders in various fields: https://www.odoo.com/odoo/project/133/tasks/4277083

**Comment by aab-odoo** [comment]
Yep, we don't do such development in long time stable branchs like 17.0, so if you really want this, you can do a custo in accounting, that could be removed in saas-18.2.

However, I changed my mind about the placeholder field to be a date field. With this version, dates aren't formatted like they should be. When I said char field, I was thinking of placeholders "explaining what the value should be", not dates directly. So if you want a date as placeholder, the placeholder field should be a date field, and properly formatted.

**Comment by RicGR98** [comment]
@aab-odoo 

>  you can do a custo in accounting, that could be removed in saas-18.2.

Will do that then :)


> dates aren't formatted like they should be

This can be taken care of in the backend while computing the placeholder, nope ? I don't really have a preference between the two versions that have been pushed but this doesn't strike as a problem to me

**Comment by aab-odoo** [comment]
For your usecase, it makes more sense to have a date field. In the master spec, we plan to support both date and char fields, but in the case of char fields we would not format the value as a date. It's meant to be used as "explanatory text".

**Comment by RicGR98** [comment]
Closing this as we're gonna wait for the standard way of doing it in https://www.odoo.com/odoo/project/133/tasks/4277083

**Pull request closed without merging**